diff -Naur ESP8266Audio/src/AudioGeneratorMIDI.cpp ESP8266Audio-1_9_5A10001986/src/AudioGeneratorMIDI.cpp
--- ESP8266Audio/src/AudioGeneratorMIDI.cpp	2021-12-18 05:15:12.000000000 +0100
+++ ESP8266Audio-1_9_5A10001986/src/AudioGeneratorMIDI.cpp	2023-05-29 01:54:06.000000000 +0200
@@ -58,6 +58,10 @@
 
 #include "AudioGeneratorMIDI.h"
 
+#if __GNUC__ == 8
+// Do not build, GCC8 has a compiler bug
+#else // __GNUC__ == 8
+
 #pragma GCC optimize ("O3")
 
 #define TSF_NO_STDIO
@@ -637,3 +641,4 @@
   afs->size = &afs_size;
 }
 
+#endif
\ No newline at end of file
diff -Naur ESP8266Audio/src/AudioGeneratorMIDI.h ESP8266Audio-1_9_5A10001986/src/AudioGeneratorMIDI.h
--- ESP8266Audio/src/AudioGeneratorMIDI.h	2021-09-16 18:33:23.000000000 +0200
+++ ESP8266Audio-1_9_5A10001986/src/AudioGeneratorMIDI.h	2023-05-29 01:54:23.000000000 +0200
@@ -21,6 +21,10 @@
 #ifndef _AUDIOGENERATORMIDI_H
 #define _AUDIOGENERATORMIDI_H
 
+#if __GNUC__ == 8
+// Do not build, GCC8 has a compiler bug
+#else // __GNUC__ == 8
+
 #include "AudioGenerator.h"
 
 #define TSF_NO_STDIO
@@ -179,3 +183,4 @@
 
 #endif
 
+#endif
diff -Naur ESP8266Audio/src/AudioOutputI2S.cpp ESP8266Audio-1_9_5A10001986/src/AudioOutputI2S.cpp
--- ESP8266Audio/src/AudioOutputI2S.cpp	2021-12-18 05:15:13.000000000 +0100
+++ ESP8266Audio-1_9_5A10001986/src/AudioOutputI2S.cpp	2023-05-29 14:24:54.000000000 +0200
@@ -98,6 +98,7 @@
 
 AudioOutputI2S::~AudioOutputI2S()
 {
+  /*
   #ifdef ESP32
     if (i2sOn) {
       audioLogger->printf("UNINSTALL I2S\n");
@@ -110,6 +111,8 @@
     stop();
   #endif
   i2sOn = false;
+  */
+  stop();
 }
 
 bool AudioOutputI2S::SetRate(int hz)
@@ -201,7 +204,11 @@
       }
       else if (lsb_justified)
       {
+#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(4, 2, 0)
+        comm_fmt = (i2s_comm_format_t) I2S_COMM_FORMAT_STAND_MSB;
+#else
         comm_fmt = (i2s_comm_format_t) (I2S_COMM_FORMAT_I2S | I2S_COMM_FORMAT_I2S_LSB);
+#endif	
       }
       else
       {
@@ -345,6 +352,10 @@
 
   #ifdef ESP32
     i2s_zero_dma_buffer((i2s_port_t)portNo);
+    audioLogger->printf("UNINSTALL I2S\n");
+    i2s_driver_uninstall((i2s_port_t)portNo); //stop & destroy i2s driver
+  #elif defined(ESP8266)
+    i2s_end();
   #elif defined(ARDUINO_ARCH_RP2040)
     I2S.end();
   #endif
diff -Naur ESP8266Audio/src/spiram-fast.h ESP8266Audio-1_9_5A10001986/src/spiram-fast.h
--- ESP8266Audio/src/spiram-fast.h	2021-09-16 18:33:23.000000000 +0200
+++ ESP8266Audio-1_9_5A10001986/src/spiram-fast.h	2023-05-29 01:39:03.000000000 +0200
@@ -332,9 +332,12 @@
             }
             digitalWrite(csPin, HIGH);
 
-            pinMode(sck, SPECIAL);
-            pinMode(miso, SPECIAL);
-            pinMode(mosi, SPECIAL);
+            //pinMode(sck, SPECIAL);
+            //pinMode(miso, SPECIAL);
+            //pinMode(mosi, SPECIAL);
+            pinMode(sck, OUTPUT);
+            pinMode(miso, OUTPUT);
+            pinMode(mosi, OUTPUT);
             pinMode(csPin, OUTPUT);
 
             // Enable streaming read/write mode
